src/Host/Client/Impl/Host/RemoteCredentialsDecorator.cs

```diff
@@ -21,6 +21,7 @@ internal class RemoteCredentialsDecorator : ICredentialsDecorator {
         private readonly AsyncReaderWriterLock _lock;
         private bool _credentialsAreValid;
         private readonly string _workspaceName;
+        private object _credentialsLock;
```

Comment:
Actually yes, I missed this in the diff-only. Doesn't `_credentialsLock` remove need for the token?